lib/src/checkbox/Checkbox.tsx

```diff
@@ -6,104 +6,110 @@ import { v4 as uuidv4 } from "uuid";
 import useTheme from "../useTheme";
 import useTranslatedLabels from "../useTranslatedLabels";
 import BackgroundColorContext from "../BackgroundColorContext";
-import CheckboxPropsType, { Margin, Space } from "./types";
+import CheckboxPropsType, { Margin, Space, RefType } from "./types";
 
 const checkedIcon = (
   <svg fill="currentColor" focusable="false" aria-hidden="true" viewBox="0 0 24 24">
     <path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"></path>
   </svg>
 );
 
-const DxcCheckbox = ({
-  checked,
-  defaultChecked = false,
-  value,
-  label = "",
-  labelPosition = "before",
-  name = "",
-  disabled = false,
-  optional = false,
-  onChange,
-  margin,
-  size = "fitContent",
-  tabIndex = 0,
-}: CheckboxPropsType): JSX.Element => {
-  const [labelId] = useState(`label-checkbox-${uuidv4()}`);
-  const [innerChecked, setInnerChecked] = useState(defaultChecked);
+const DxcCheckbox = React.forwardRef<RefType, CheckboxPropsType>(
+  (
+    {
+      checked,
+      defaultChecked = false,
+      value,
+      label = "",
+      labelPosition = "before",
+      name = "",
+      disabled = false,
+      optional = false,
+      onChange,
+      margin,
+      size = "fitContent",
+      tabIndex = 0,
+    },
+    ref
+  ) => {
```

Comment:
Add `: JSX.Element` to the return value type.

Reply:
Done